src/bpmncwpverify/core/spin.py

```diff
@@ -124,7 +124,7 @@ def with_behavior_(
             return builder_result
         if not_(is_successful)(behavior_str):
             error = unsafe_perform_io(behavior_str.failure())
-            return Failure(ExceptionError(str(error)))
+            return Failure(error)
```

Comment:
I'm looking at this code more closely only because I am work on #70 that moves out the `IOResultE` from the `spin.Builder` code. If you unwrap the `Failure` type, then you end up with something of type `Exception`, and that  should fail type checking unless because `Exception` is not a subtype of `Error`. Right?

We need to create a `FailedToReadFile` error type that can be build from the exception? Or maybe in this case, the `ExceptionError` is the right thing to do only because it is a result of an IO operation?